conn.go

```diff
@@ -11,7 +12,7 @@ import (
 )
 
 type Conn struct {
-	inet.Stream
+	stream inet.Stream
```

Comment:
I don't think you need to remove the embedding. You should be able to override the `Close()` method delegating the call to `c.Stream.Close()`.

Reply:
This is also narrowing the interface to that of a `Conn`, so it's not the same.